replay/models/base_rec.py

```diff
@@ -1694,14 +1694,17 @@ def _predict_without_sampling(
             ),
         )
 
-        max_hist_len = (
-            self._calc_max_hist_len(log, users)
-            if filter_seen_items and log is not None
-            else 0
-        )
+        if filter_seen_items and log is not None:
+            user_to_num_items = log.groupBy("user_idx").agg(
+                sf.countDistinct("item_idx").alias("num_items")
+            )
+            users = users.join(user_to_num_items, on="user_idx")
+            return users.join(
+                selected_item_popularity, on=(sf.col("rank") <= k + sf.col("num_items")), how="left"
```

Comment:
you can also all filter here as below to avoid coping of entire table. did you measure computing time in this case?